cpp/roaring64map.hh

```diff
@@ -1051,6 +1051,26 @@ public:
         return result;
     }
 
+    /**
+     * Return the number of bytes of memory used by this bitmap
+     */
+    size_t getMemorySizeInBytes() const {
+        // Figuring out how much memory is used by a std::map is guesswork.
+        // A common red/black tree implementation has 3 pointers plus 2 ints
+        // per element, plus the size of the pair.  The size of the Roaring
+        // struct is included in roarings.getMemorySizeInBytes() so remove it.
+        constexpr size_t perEntry = 3 * sizeof(void*) + 2 * sizeof(int) + sizeof(std::pair<uint32_t, Roaring>) - sizeof(Roaring);
```

Comment:
I agree that counting bytes is hard. I wonder if this calculation has to have an accommodation for the "new" header... since map nodes are relatively small, the amount of header added by the storage allocator could be significant. I would imagine it's the size of a couple of pointers at least but I don't actually know.

Reply:
Yes, deciding exactly what to count is not always simple.  Do you count the header?  Do you count all the allocated bytes not just the "usable" bytes (often allocators will give a chunk of memory of a set size rather than the exact number of bytes)?  These things are essentially not countable, at least not in a portable way (some systems have malloc_usable_size() which returns the latter value).  I don't know that the first value is of much interest in general although of course if you are allocating tons of small things, it could be non-trivial.  In general in that situation you'd probably move to something like jemalloc which uses a very different allocation model than the "traditional" one of allocating n+X bytes and returning a pointer into the buffer.